### DISCUSSION TITLE
Option-B for deno compile: is it being worked on?

### DISCUSSION THREAD

**Discussion by atzlt** [denoland/deno#18798] [General]
As mentioned in [this comment](https://github.com/denoland/deno/issues/986#issuecomment-729077429), there are two approaches two compile a deno file into binary (correct me if I'm wrong):

1. Include a copy, at least the required parts, of the deno executable, and pack the source code into the executable.
2. Create a Rust project from the source code, and build it using Rust.

It seems like we're using the first approach, and in the issue mentioned before [the second approach is being worked on](https://github.com/denoland/deno/issues/986#issuecomment-735997439). I would like to ask the progress.

Actually there are quite a few disadvantages of the first approach. First the emitted binary is large (~70 MB for a `console.log("Hello, world!")`). Second the source code is directly available through the binary (it's included at the end of the binary in plain text!). The second approach seems promising to solve these two problems.

**Comment by bartlomieju** [comment]
There has been no progress on the Option B, the core team currently doesn't have the capacity to look into it.

I'll be happy to provide pointers how option B could be implemented starting as a community project.
